frontend/src/pages/HomePage.tsx

```diff
@@ -18,10 +19,21 @@ export const HomePage = () => {
       console.log('logging in', response.idToken);
 
       // TODO: Make request to /userLoginSignup using response.idToken
-
-      // user variable is placeholder for data which will be returned from BE
-      const user = { name: 'John Smith', faculty: '' };
-      user.faculty ? navigate('/chats') : navigate('/update-user');
+      try {
+        const res = await http.post('/userLoginSignup', undefined, {
```

Comment:
There is a small period of time where the user has been successfully redirected back but this network request to the backend is still in progress. For the user, it looks like nothing is happening. We should fix this by adding a [loading spinner](https://mui.com/material-ui/react-progress/) at the start of the signInSignUp function and then stop it once we error or have successfully made the request. 

Note: on local the network request happens pretty quickly but if you want to test this, you can just temporarily comment out the navigate